docs/UML/classes/org.openehr.am.aom2.adl_code_definitions.adoc

```diff
@@ -1,9 +1,96 @@
 === ADL_CODE_DEFINITIONS Class
 
+ADL2 was initially released with code system using id-codes (for nodes), at-codes (for values only) and ac-codes (for value sets). As this proved to hinder its widespread use in the openEHR community (due to its impact on existing openEHR RM data) a code system matching the one used in ADL1.4 was introduced as openEHR primary code system.
+
+openEHR systems must support at least at-coded ADL2 archetypes.
+
+[cols="^1,3,5"]
+|===
+h|*Class*
+2+^h|*ADL_CODE_DEFINITIONS* (at-coded ADL2)
+
+h|*Description*
+2+a|Definitions relating to the internal code system of archetypes.
+
+h|*Constants*
+^h|*Signature*
+^h|*Meaning*
+
+h|*1..1*
+|*At_code_leader*: `String{nbsp}={nbsp}"at"`
+a|String leader of ‘identifier’ codes, i.e. codes used to identify archetype nodes.
+
+h|*1..1*
+|*Value_code_leader*: `String{nbsp}={nbsp}"at"`
+a|String leader of ‘value’ codes, i.e. codes used to identify codes values, including value set members.
+
+h|*1..1*
+|*Value_set_code_leader*: `String{nbsp}={nbsp}"ac"`
+a|String leader of ‘value set’ codes, i.e. codes used to identify value sets.
+
+h|*1..1*
+|*Specialisation_separator*: `char{nbsp}={nbsp}'.'`
+a|Character used to separate numeric parts of codes belonging to different specialisation levels.
+
+h|*1..1*
+|*Code_regex_pattern*: `String{nbsp}={nbsp}"(0&#124;[1-9][0-9]&#42;)(\.(0&#124;[1-9][0-9]&#42;))&#42;"`
+a|Regex used to define the legal numeric part of any archetype code. Corresponds to the simple pattern of dotted numbers, as used in typical multi-level numbering schemes.
+
+h|*1..1*
+|*Root_code_regex_pattern*: `String{nbsp}={nbsp}"^at0000(\.1)&#42;$"`
```

Comment:
@joostholslag 
discuss how to handle **at-coded** vs **id-coded** tables under https://specifications.openehr.org/releases/AM/development/AOM2.html#_adl_code_definitions_class considering backwards compatibility

Reply:
I would suggest to just keep all in the same class to keep backwards compatibility. The regex for the root node would then simply accept either id1(.1)* or at0000(.1)*. Also in Archie we are just planning to let the validator simply check if all id-codes are either at-codes or id-codes.

Reply:
Fixed in https://github.com/openEHR/specifications-AM/commit/7fa959c109f8b116f91b16763a48e594ab425431